### PR TITLE
Integrate `isort`

### PR DESCRIPTION
**Pull request by RomiPolaczek**
### **Goal**

Arrange Fuse's code base, and particular the imports in each file.
**Related issue:** #214

### **Key files**:

- `.pre-commit-config.yaml`: Add `isort` as a pre-commit hook to enforce consistent import order.
- `/.github/workflows/lint.yaml`: Added `isort` into Github actions for automated linting during CI.
- `setup.cfg`: Configured `isort` to use the `black` profile, ensuring alignment between import sorting and code formatting standards.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -25,3 +25,7 @@ repos:
         args:
           - --remove-empty-cells
           - --preserve-cell-outputs
+  - repo: https://github.com/pycqa/isort
```
Comment (mosheraboh) [inline .pre-commit-config.yaml:28]: I assume this step checks and applies fixes if necessary, right?
Is the order important? meaning if black should run before or after.

**Review by mosheraboh — APPROVED** [review]
Hi @RomiPolaczek ✋ 
Great work!
Approving with one question inline 
(making sure that the precommit step both checks and applies fixes)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (RomiPolaczek) [reply .pre-commit-config.yaml:24]: Hi @mosheraboh,

Thank you for the feedback! 🙌

Yes, the pre-commit step for `isort` both checks and applies fixes as necessary to ensure imports are properly sorted.
As for the order: It's generally recommended that `isort` runs before black, as `isort`organizes imports and `black` formats the entire file. This avoids potential conflicts where `black` might undo some of `isort`'s changes.

I’ve now pushed a commit that updates the order in `.pre-commit-config.yaml` to ensure `isort` runs before `black`.

Comment (mosheraboh) [reply .pre-commit-config.yaml:24]: 🙏

**Review by mosheraboh — APPROVED** [review]
Great! Thanks again.

**Pull request merged**